tools/vswhere/MSBuildLocator.cs

```diff
@@ -0,0 +1,71 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+
+namespace Xamarin.Android.Tools.VSWhere
+{
+	/// <summary>
+	/// https://github.com/Microsoft/vswhere/wiki/Find-MSBuild
+	/// </summary>
+	public class MSBuildLocator
+	{
+		static readonly string [] msbuildLocations = new [] {
+			// VS 2019 & Higher
+			Path.Combine ("MSBuild", "Current", "Bin", "MSBuild.exe"),
+			// VS 2017
+			Path.Combine ("MSBuild", "15.0", "Bin", "MSBuild.exe"),
+		};
+
+		/// <summary>
+		/// Adds the -prerelease flag, defaults to false
+		/// </summary>
+		public bool IncludePrerelease { get; set; } = false;
+
+		public string VisualStudioDirectory { get; private set; }
+
+		public string MSBuildPath { get; private set; }
+
+		public void Locate ()
```

Comment:
Should this instead be an `MSBuildLocator` constructor?  This API usage seems weird:

```csharp
var locator = new MSBuildLocator();
locator.Locate();
// use locator.VisualStudioDirectory or locator.MSBuildPath ...
```

I suppose the benefit is around the `IncludePrerelease` property -- it would otherwise need to be a constructor parameter -- and it's versional this way.  It just seems...odd.